domains/transfer/Transfer.API/API/Transfer/Api/Dto/Requests/CreateTransferAgreementProposal.cs

```diff
@@ -1,42 +1,41 @@
-using System;
-using API.Shared.Extensions;
 using FluentValidation;
 using Microsoft.AspNetCore.Http;
+using System;
+using API.Shared.Extensions;
 
 namespace API.Transfer.Api.Dto.Requests;
 
-public class CreateTransferAgreementValidator : AbstractValidator<CreateTransferAgreement>
+public record CreateTransferAgreementProposal(long StartDate,
+    long? EndDate,
+    string ReceiverTin);
+
+public class CreateTransferAgreementProposalValidator : AbstractValidator<CreateTransferAgreementProposal>
 {
-    public CreateTransferAgreementValidator(IHttpContextAccessor context)
+    public CreateTransferAgreementProposalValidator(IHttpContextAccessor context)
     {
         var now = DateTimeOffset.UtcNow;
 
-        RuleFor(createTransferAgreement => createTransferAgreement.StartDate)
+        RuleFor(createInvitation => createInvitation.StartDate)
             .NotEmpty()
             .WithMessage("Start Date cannot be empty.")
             .GreaterThanOrEqualTo(_ => now.ToUnixTimeSeconds())
             .WithMessage("Start Date cannot be in the past.")
             .MustBeBeforeYear10000();
 
-        RuleFor(createTransferAgreement => createTransferAgreement.EndDate)
+        RuleFor(createInvitation => createInvitation.EndDate)
             .Cascade(CascadeMode.Stop)
-            .Must((createTransferAgreement, endDate) => endDate == null || endDate > createTransferAgreement.StartDate)
+            .Must((createInvitation, endDate) => endDate == null || endDate > createInvitation.StartDate)
```

Comment:
Suggestion to rename the variable to createProposal

Reply:
corrected